src/sonic-yang-mgmt/tests/yang-model-tests/yangModelTesting.py

```diff
@@ -0,0 +1,426 @@
+            self.ctx = ly.Context(yangDir)
+            log.debug(yangFiles)
+            for f in yangFiles:
+                # load a module
+                log.debug(f)
+                m = self.ctx.get_module(f)
+                if m is not None:
+                    log.error("Could not get module: {}".format(m.name()))
+                else:
+                    m = self.ctx.load_module(f)
+                    if m is not None:
+                        log.info("module: {} is loaded successfully".format(m.name()))
+                    else:
+                        return
+        except Exception as e:
+            printExceptionDetails()
+            raise e
+        return
+
+    """
+        Run all tests from list self.tests
+    """
+    def run(self):
+        try:
+            self.loadYangModel(self.yangDir)
+            ret = 0
+            for test in self.tests:
+                func = "yangTest" + test;
+                # Pass function name, needed to retrieve input Json Dictionary
+                ret = ret + getattr(self, func)(func)
+        except Exception as e:
+            printExceptionDetails()
+            raise e
+        return ret
+
+    """
+        Get the JSON input based on func name
+        and return jsonInput
+    """
+    def readJsonInput(self, func):
+        try:
+            # load test specific Dictionary, using Key = func
+            # this is to avoid loading very large JSON in memory
+            log.debug(" Read JSON Section: " + func)
+            jInput = ""
+            with open(self.jsonFile, 'rb') as f:
+                jInst = ijson.items(f, func)
+                for it in jInst:
+                    jInput = jInput + json.dumps(it)
+            log.debug(jInput)
+        except Exception as e:
+            printExceptionDetails()
+        return jInput
+
+    """
+        Log the start of a test
+    """
+    def logStartTest(self, tStr):
+        log.info("\n------------------- Test "+ str(self.testNum) +\
+        ": " + tStr + "---------------------")
+        self.testNum = self.testNum + 1
+        return
+
+    """
+        Load Config Data and return Exception as String
+    """
+    def loadConfigData(self, jInput):
+        s = ""
+        try:
+            node = self.ctx.parse_data_mem(jInput, ly.LYD_JSON, \
+            ly.LYD_OPT_CONFIG | ly.LYD_OPT_STRICT)
+        except Exception as e:
+            s = str(e)
+            log.debug(s)
+        return s
+
+    """
+        Test 1: Configure Wrong family with ip-prefix for VLAN_Interface Table.
+    """
+    def yangTest1(self, func):
+        try:
+            tStr = " Configure Wrong family with ip-prefix for VLAN_Interface Table."
+            self.logStartTest(tStr)
+            jInput = self.readJsonInput(func)
+            # load the data, expect a exception with must condition failure
+            s = self.loadConfigData(jInput)
+            if ("Must condition" in s and "not satisfied" in s):
```

Comment:
I think the error strings from libyang should be defined separately as a map or something. 

More over, if possible, we could make the tStr be part of the yangtest.json or a separate map, and so we could have a common test function called by different test cases.

Reply:
Done.